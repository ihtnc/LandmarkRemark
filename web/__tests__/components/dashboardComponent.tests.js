import React from 'react';
import { shallow } from 'enzyme';
import Dashboard from "@components/dashboardComponent";

describe("LandmarkMapComponent", () => {
  test("should render correctly on ready state", () => {
    const email = 'email';

    const wrapper = shallow(<Dashboard email={email} />)
    expect(wrapper).toMatchSnapshot();
  });

  test("should render correctly when there is an error", () => {
    const email = 'email';
    const status = 'message';
    const error = true;

    const wrapper = shallow(<Dashboard email={email} status={status} error={error} />)
    expect(wrapper).toMatchSnapshot();
  });

  test("should render correctly when there is a status update", () => {
    const email = 'email';
    const status = 'message';
    const error = false;

    const wrapper = shallow(<Dashboard email={email} status={status} error={error} />)
    expect(wrapper).toMatchSnapshot();
  });
});