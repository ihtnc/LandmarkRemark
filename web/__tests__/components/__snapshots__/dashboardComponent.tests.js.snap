// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandmarkMapComponent should render correctly on ready state 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      Email:
    </styled.div>
    <styled.div>
      email
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      Filter:
    </styled.div>
    <styled.input
      onChange={[Function]}
      value=""
    />
    <styled.button
      disabled={false}
      onClick={[Function]}
    >
      Apply
    </styled.button>
  </styled.div>
  <styled.div>
    <styled.button
      disabled={false}
      onClick={[Function]}
    >
      Log Out
    </styled.button>
    <styled.span
      error={false}
    />
  </styled.div>
  <div>
    <styled.div>
      <styled.span
        expand={true}
        onClick={[Function]}
      />
      <div
        onClick={[Function]}
      >
        Click to view instructions
      </div>
    </styled.div>
  </div>
</styled.div>
`;

exports[`LandmarkMapComponent should render correctly when there is a status update 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      Email:
    </styled.div>
    <styled.div>
      email
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      Filter:
    </styled.div>
    <styled.input
      onChange={[Function]}
      value=""
    />
    <styled.button
      disabled={false}
      onClick={[Function]}
    >
      Apply
    </styled.button>
  </styled.div>
  <styled.div>
    <styled.button
      disabled={false}
      onClick={[Function]}
    >
      Log Out
    </styled.button>
    <styled.span
      error={false}
    >
      message
    </styled.span>
  </styled.div>
  <div>
    <styled.div>
      <styled.span
        expand={true}
        onClick={[Function]}
      />
      <div
        onClick={[Function]}
      >
        Click to view instructions
      </div>
    </styled.div>
  </div>
</styled.div>
`;

exports[`LandmarkMapComponent should render correctly when there is an error 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      Email:
    </styled.div>
    <styled.div>
      email
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      Filter:
    </styled.div>
    <styled.input
      onChange={[Function]}
      value=""
    />
    <styled.button
      disabled={false}
      onClick={[Function]}
    >
      Apply
    </styled.button>
  </styled.div>
  <styled.div>
    <styled.button
      disabled={false}
      onClick={[Function]}
    >
      Log Out
    </styled.button>
    <styled.span
      error={true}
      show={true}
    >
      message
    </styled.span>
  </styled.div>
  <div>
    <styled.div>
      <styled.span
        expand={true}
        onClick={[Function]}
      />
      <div
        onClick={[Function]}
      >
        Click to view instructions
      </div>
    </styled.div>
  </div>
</styled.div>
`;
